sdv/data_processing/data_processor.py

```diff
@@ -575,7 +582,10 @@ def prepare_for_fitting(self, data):
             self._fit_formatters(data)
 
             LOGGER.info(f'Fitting constraints for table {self.table_name}')
-            constrained = self._fit_transform_constraints(data)
+            if len(self._constraints_list) != len(self._constraints):
```

Comment:
maybe we should add a unit test for this case

Reply:
This test covers it [here](https://github.com/sdv-dev/SDV/pull/1429/files#diff-65e7498d86df62640dd967d34beb38e7db5acaf9b73a2e240158313e61c08328R1300-R1312) ps: Scroll down and you will see it highlighted.
```python
        dp._transform_constraints.return_value = transformed_data
        dp._prepared_for_fitting = False
        dp._hyper_transformer.get_config.return_value = {'sdtypes': {}, 'transformers': {}}
        dp._constraints_list = [1, 2, 3] 
        dp._constraints = []

        # Run
        DataProcessor.prepare_for_fitting(dp, data)

        # Assert
        pd.testing.assert_series_equal(dp._dtypes, pd.Series([np.int64], index=['a']))
        dp._transform_constraints.assert_called_once_with(data)
        dp._fit_constraints.assert_called_once_with(data)
```